Quotient/connection.cpp

```diff
@@ -58,6 +58,15 @@
 #include <QtCore/QStringBuilder>
 #include <QtNetwork/QDnsLookup>
 
+#include <openssl/aes.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include <openssl/kdf.h>
+#include <openssl/rand.h>
+#include <openssl/sha.h>
```

Comment:
Can we have all those OpenSSL wrappers in a separate h/cpp file pair? I'd rather we try hard to reduce the size of `connection.cpp` and `room.cpp`. These things are quite self-contained and don't need Quotient context.

Btw, did you check the Qt(6) API, if they already wrapped some of these, by chance? Just asking so that I don't re-do what you likely already did.